port/unix/omrshmem.c

```diff
@@ -1663,7 +1663,7 @@ omrshmem_openSharedMemory (OMRPortLibrary *portLibrary, intptr_t fd, const char
 				goto failDontUnlink;
 			}
 		} else {
-#if defined(__GNUC__) || defined(AIXPPC)
+#if (defined(__GNUC__) || defined(AIXPPC)) && !defined(J9ZOS390)
```

Comment:
Suggest reordering the `#if` branches so it's clear which one is intended for z/OS, or `#define KEY` appropriately for each platform/compiler and then use `KEY` instead of `key`, `_key` or `__key`.

Reply:
can you detail that a bit, looking at the logic again, since I have excluded via `!defined(J9ZOS390)` , it should mean that z/OS does not use any of the child `if` branches underlying within the parent `if` clause right?

Reply:
I see now that none of the branches would apply for z/OS: is that the right behavior?
If so, I suggest we reorder the code to put the z/OS path first, and this in the `#else` block:
```c
#if defined(OMRZOS390)
    // existing lines 1686-1709
#elif defined(__GNUC__) || defined(AIXPPC)
    // existing lines 1667-1683
#endif /* defined(OMRZOS390) */
```

Reply:
yup, and that does look cleaner.
But what is the difference (if any) between `OMRZOS390` and `J9ZOS390` ... because if it seems there is diff, then I will still need it as follows:
```
#if defined(OMRZOS390)
    // existing lines 1686-1709
#elif (defined(__GNUC__) || defined(AIXPPC)) && !defined(J9ZOS390)
    // existing lines 1667-1683
#endif /* defined(OMRZOS390) */
```

Reply:
I think `OMRZOS390` is the preferred flag to test in new code, `J9ZOS390` is a name that originated in OpenJ9. They're both still used and I think they're equivalent, but OMR is supposed to be a base for OpenJ9, not dependent on anything from OpenJ9 (including naming conventions).

Reply:
Seems the build is failing after reorganizing this and `omrshsem_deprecated.c` 
```
cd .../omr/build/port && /xlc210/usr/lpp/IBM/cnw/v2r1/openxl/bin/ibm-clang64  -D__STDC_LIMIT_MACROS -D_ALL_SOURCE -D_OPEN_THREADS=3 -D_XOPEN_SOURCE=600 -DJ9ZOS390 -DJ9ZOS39064 -DLONGLONG -DOMR_EBCDIC -DOMRPORT_LIBRARY_DEFINE -DSUPPORTS_THREAD_LOCAL -DZOS -I...(all includes here)  -fstrict-aliasing -mzos-target=ZOSV2R4 -m64 -march=arch10   -fvisibility=default -o CMakeFiles/omrport_obj.dir/unix/omrshmem.c.o   -c /.../omr/port/unix/omrshmem.c  
```

The full command follows the above format, and `J9JZ0S390` is defined, but looks like `OMRZOS390` is not ... so I might need to switch it to `J9JZOS390`. With that, I think the reorganizing should work fine.

Reply:
I don't see `OMRZOS390` defined anywhere (despite tests in this file and in `omrsharedhelper.h`).
Unless someone can point to where `OMRZOS390` is defined, this should use `J9ZOS390`.

Reply:
Agreed, thats what I was mentioning in my prior comment. As such, in the current state of the PR I am still utilizing `J9ZOS390` in this file.